src/main/java/com/example/demo/services/ImageService.java

```diff
@@ -0,0 +1,147 @@
+package com.example.demo.services;
+
+import com.example.demo.models.Image;
+import com.example.demo.models.Label;
+import com.example.demo.repository.ImageRepository;
+import com.example.demo.repository.LabelRepository;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import org.json.simple.JSONObject;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ImageService {
+    private final ImageRepository imageRepository;
+    private final LabelRepository labelRepository;
+    private  final ImaggaAPI imaggaAPI;
+
+    public ImageService(ImageRepository imageRepository, LabelRepository labelRepository, ImaggaAPI imaggaAPI) {
+        this.imageRepository = imageRepository;
+        this.labelRepository = labelRepository;
+        this.imaggaAPI = imaggaAPI;
+    }
+
+    public String uploadImage(Image image) {
+        String imageUrl = image.getUrl();
+        String jsonResponse = imaggaAPI.categorizeImage(imageUrl);
```

Comment:
Instead of `imaggaAPI.categorizeImage` to return a `String` we can use the features of `WebClient` and convert it to automatically return instances of the Java classes which we create below by hand using `ObjectMapper`. If you use that feature of automatic deserialization of `WebClient` the code below will become a lot shorter since you won't need to use `ObjectMapper` at all yourself.